crate2nix/templates/nix/crate2nix/default.nix

```diff
@@ -70,65 +69,88 @@ rec {
     );
 
   /* A restricted overridable version of  buildRustCrateWithFeaturesImpl. */
-  buildRustCrateWithFeatures = {
-        packageId, 
-        features ? rootFeatures,
-        crateOverrides ? defaultCrateOverrides, 
-        buildRustCrateFunc ? buildRustCrate
-      }:
+  buildRustCrateWithFeatures =
+    { packageId
+    , features ? rootFeatures
+    , crateOverrides ? defaultCrateOverrides
+    , buildRustCrateFunc ? buildRustCrate
+    , doTest ? false,
+    }:
     lib.makeOverridable
-      ({features, crateOverrides}: 
+      ({features, crateOverrides, doTest}:
         let builtRustCrates = builtRustCratesWithFeatures {
-          inherit packageId features crateOverrides  buildRustCrateFunc;
+          inherit packageId features crateOverrides buildRustCrateFunc doTest;
         };
         in builtRustCrates.${packageId})
-      { inherit features crateOverrides; };
+      { inherit features crateOverrides doTest; };
 
   /* Returns a buildRustCrate derivation for the given packageId and features. */
-  builtRustCratesWithFeatures = { 
-        crateConfigs? crates, 
-        packageId,
-        features,
-        crateOverrides, 
-        buildRustCrateFunc
-      } @ args:
+  builtRustCratesWithFeatures =
+    { packageId
+    , features
+    , crateConfigs ? crates
+    , crateOverrides
+    , buildRustCrateFunc
+    , doTest
+    , target ? defaultTarget
+    } @ args:
     assert (builtins.isAttrs crateConfigs);
     assert (builtins.isString packageId);
     assert (builtins.isList features);
+    assert (builtins.isAttrs target);
```

Comment:
What about doTest?

I also think that deserves some mention in the function comment. Is it a bool? What does it enable?

Also, is it a customary name for running tests in nix that other build generators use? Otherwise, I'd prefer `runTests`.

Reply:
In nixpkgs (and stdenv) there are a few "phases" that match what we are going to do here:

- checkPhase
- installCheckPhase

and the boolean flag for that is something like `doCheck`.

I have no opinions on this option. I think going with something around `check` is probably best if we aim for nixpkgs compatibility.